### ISSUE TITLE
NameError: name 'flagger_process' is not defined

### ISSUE THREAD

**Issue by astrosumit** [daniellefenech/SERPent#5]
Hi,
I was trying to run SERPent.py on a uvdata file 'J0741+2706_X_1996_05_15_yyk_vis.fits' downloaded from astrogeo.org.

i used the option use_fits = 'no'. On running the SERPent program, it says:
NameError: name 'flagger_process' is not defined

Then I used the option use_fits = 'yes'. It then shows different errors:
-----------------------------------------------------------------------
finished alt writing flag table in 0.00200605392456
src is []
[]
Flagged freq: 0
JOB 2 of 140:  Nothing flagged. Skipping run...
JOB 2 of 140:  Nothing flagged. Skipping run...
no
flag cols
/home/sumit/.local/lib/python2.7/site-packages/numpy/lib/function_base.py:3405: RuntimeWarning: Invalid value encountered in median
  r = func(a, **kwargs)
JOB 3 of 140:  Max value < kickout. Skipping run...
JOB 3 of 140:  Max value < kickout. Skipping run...
no
flag cols
Flagged freq: 0
finished alt writing flag table in 0.00240588188171
-----------------------------------------------------------------------
FITLD1: Task FITLD  (release of 31DEC18) begins
FITLD1: Found SOURCE   observed on 15-MAY-1996
FITLD1: Create SOURCE      .SERP  .   1 (UV)  on disk  1  cno    3
FITLD1: UVFDAT: counts for auto-correlations
FITLD1: UVFDAT: found          1 spectra samples already flagged
FITLD1: UVFDAT: found    100.000 percent samples already flagged
FITLD1: UVFDAT: omitted          1 totally flagged vis records
FITLD1: UVPGET RANDOM PARMS MISSING: -1 -1 -1  3 -1  5  6  7  8  4  9 10
FITLD1: UVPGET: Missing COMPLEX  axis
FITLD1: RNXINI: DATA NOT IN TIME ORDER => NO NX TABLE
FITLD1: NXCHCK: ERROR   1 ON INIT INDEX TABLE (RE)WRITE
FITLD1: Image=SOURCE    (UV)         Filename=SOURCE      .SERP  .   1
FITLD1: Telescope=e-MERLIN           Receiver=VLBA
FITLD1: Observer=Calibrat            User #=   11
FITLD1: Observ. date=15-MAY-1996     Map date=08-JUN-2020
FITLD1: ----------------------------------------------------------------
FITLD1: Type    Pixels   Coord value     at Pixel     Coord incr   Rotat
FITLD1:              3   1.0000000E+00       1.00  1.0000000E+00    0.00
FITLD1:              4  -1.0000000E+00       1.00 -1.0000000E+00    0.00
FITLD1:            512   8.1504900E+09       1.00  8.0000000E+06    0.00
FITLD1:              8   1.0000000E+00       1.00  1.0000000E+00    0.00
FITLD1:              1   1.1535722E+02       1.00  1.0000000E+00    0.00
FITLD1:              1   2.7112609E+01       1.00  1.0000000E+00    0.00
FITLD1: ----------------------------------------------------------------
FITLD1: Observed RA   07 41 25.733    DEC  27 06 45.39
FITLD1: Maximum version number of extension files of type HI is   1
FITLD1: Maximum version number of extension files of type FG is   1
FITLD1: Appears to have ended successfully
FITLD1: localhost    31DEC18 TST: Cpu=       0.0  Real=       0
Obit: warn    20200608T234450 Could not find AIPS MA file SOURCE       SERP   seq 1 disk 1
Obit: Error   20200608T234450 ERROR opening file /home/sumit/Radio/AIPS/DATA/LOCALHOST_1/MADBAD001.00B;
Obit: Error   20200608T234450  This occured at file ObitFile.c: line 394 (ObitFileOpen)
Obit: Error   20200608T234450 No such file or directory
Obit: Error   20200608T234450  This occured at file ObitFile.c: line 1087 (ObitFileErrMsg)
Obit: trace   20200608T234450 Traceback: routine ObitIOImageAIPSOpen:  object SOURCE
Obit: trace   20200608T234450  at file ObitIOImageAIPS.c: line 410 (ObitIOImageAIPSOpen)
Obit: trace   20200608T234450 Traceback: routine ObitImageOpen:  object SOURCE
Obit: trace   20200608T234450  at file ObitImage.c: line 1064 (ObitImageOpen)
Error creating AIPS Image object
Can't find file
SOURCE
TACOP1: Task TACOP  (release of 31DEC18) begins
TACOP1: Copied FG file from vol/cno/vers  1    3   1 to  1    2   1
TACOP1: TACOP  / Copied      1 extension files
TACOP1: Appears to have ended successfully
TACOP1: localhost    31DEC18 TST: Cpu=       0.0  Real=       0
|
---------------------------------------------

Finished on 08-Jun-20 at: 23:44:50 
Time taken (hh:mm:ss): 00:00:4.85 

Final flag table is: 1

---------------------------------------------

Could you please suggest me how to resolve this issue?

Thanks in advance..

**Comment by daniellefenech** [comment]
Thanks for getting in touch and I'm sorry you're experiencing issues. On the face of it, I'm not totally sure where the problem is arising, though it looks like your source list is empty, suggesting that it may be struggling because it may not have flagged anything. Are you able to send me a copy of the inputs file you're using and point me to a copy of the data so I can investigate it further?

**Comment by astrosumit** [comment]
[test_surpent.zip](https://github.com/daniellefenech/SERPent/files/4765719/test_surpent.zip)
Hi,
Thanks for the reply. Just extract the zip file and run the command: ParselTongue edit_surpent.py 

Even if the SERPent is not finding any RFI in a data, it should not fail but generate a empty FG table. Please suggest me to solve the issue.

**Comment by daniellefenech** [comment]
Thanks for providing the test script. As hoped, the code wasn't failing due to an empty flag table (as stated, this shouldn't cause failure but the production of an empty FG table). The use_fits = 'yes' option was failing as the construction of the empty fits file incurred errors that caused problems with parseltongue. This was due to updates in astropy fits keywords. The use_fits = 'no' option was failing because of a bug where it was being passed to an incorrect function name. Your test script now runs with no problems for me with either option and produced the same result. I've pulled the changes to the master, so you'll need to update your local copy of SERPent.py from there. Do let me know if you have any more problems or questions.
